File: ChecklistSPA/src/app/models/UserChecklist.ts
import { IUserChecklist } from '../interfaces/IUserChecklist';
import { ChecklistField } from '../interfaces/IChecklistField';
import { IChecklistImage } from '../interfaces/IChecklistImage';

export class UserChecklist implements IUserChecklist {
    id: string;
    name: string;
    fields: ChecklistField[];
    createdDate: string;
    checklistImages: IChecklistImage[] = [];
}

File: ChecklistAPI/Checklist.DataLogic/Entities/UserChecklist.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Checklist.DataLogic.Entities
{
    public class UserChecklist:BaseEntity
    {
        public string Name { get; set; }
        public List<ChecklistField> Fields { get; set; }
        public DateTime CreatedDate { get; set; }
        public User User { get; set; }
    }
}
